index.js

```diff
@@ -305,6 +306,20 @@ async function fastifyHttpProxy (fastify, opts) {
     wsProxy = setupWebSocketProxy(fastify, opts, rewritePrefix)
   }
 
+  function extractUrlComponents (urlString) {
+    const [path, queryString] = urlString.split('?')
```

Comment:
I was going to merge the PR when this hit me, we recently started to use the second parameter of the split method to enforce the number of expected returned elements for performance reasons

Could you please add it as well?

```suggestion
    const [path, queryString] = urlString.split('?', 2)
```

Reply:
Sounds good, should be add in now

Reply:
Thanks!